File: go/tests/accessibility_board_test.go
// accessibility_board_test.go
//
// Verifies the CircleAI.Accessibility port (accessibility_board.go): profile
// set/get and the ordered adaptation-hint derivation (contrast, motion, aria,
// text-scale "F2", then one hint per need).

package circleai_test

import (
	"testing"

	circleai "github.com/bhengubv/CircleAI/go"
)

func TestAccessibility_HintsFor(t *testing.T) {
	b := circleai.NewInMemoryAccessibilityBoard()
	b.SetProfile(circleai.UserAccessibilityProfile{
		UserId:        "u1",
		Needs:         []circleai.AccessibilityNeed{circleai.AccessibilityNeedVisual, circleai.AccessibilityNeedMotor},
		TextScale:     1.5,
		HighContrast:  true,
		ReducedMotion: true,
		ScreenReader:  true,
	})
	if got, ok := b.GetProfile("u1"); !ok || got.TextScale != 1.5 {
		t.Fatalf("get profile = %+v ok=%v", got, ok)
	}

	hints := b.HintsFor("u1")
	// Expected fixed order: contrast, motion, aria, text-scale, need(Visual), need(Motor).
	want := []circleai.AdaptationHint{
		{Kind: "contrast", Value: "high"},
		{Kind: "motion", Value: "reduced"},
		{Kind: "aria", Value: "verbose"},
		{Kind: "text-scale", Value: "1.50"},
		{Kind: "need", Value: "Visual"},
		{Kind: "need", Value: "Motor"},
	}
	if len(hints) != len(want) {
		t.Fatalf("hint count = %d, want %d: %+v", len(hints), len(want), hints)
	}
	for i := range want {
		if hints[i] != want[i] {
			t.Fatalf("hint[%d] = %+v, want %+v", i, hints[i], want[i])
		}
	}
}

func TestAccessibility_NoProfileAndMinimal(t *testing.T) {
	b := circleai.NewInMemoryAccessibilityBoard()
	if h := b.HintsFor("nobody"); len(h) != 0 {
		t.Fatalf("hints for unknown user must be empty: %+v", h)
	}
	// TextScale == 1 emits no text-scale hint; only ScreenReader set.
	b.SetProfile(circleai.UserAccessibilityProfile{UserId: "u2", TextScale: 1.0, ScreenReader: true})
	h := b.HintsFor("u2")
	if len(h) != 1 || h[0].Kind != "aria" {
		t.Fatalf("minimal hints failed: %+v", h)
	}
}

File: typescript/tests/accessibility_board.test.ts
// accessibility_board.test.ts
// Verifies the CircleAI.Accessibility port: profile storage and the exact,
// ordered adaptation-hint derivation (including F2 text-scale and need names).

import { describe, it } from "node:test";
import assert from "node:assert/strict";
import {
  InMemoryAccessibilityBoard,
  AccessibilityDomainContext,
  AccessibilityNeed,
  userAccessibilityProfile,
} from "../src/accessibility/index";

describe("InMemoryAccessibilityBoard", () => {
  it("returns [] when no profile is set", () => {
    const b = new InMemoryAccessibilityBoard();
    assert.deepEqual(b.hintsFor("nobody"), []);
  });

  it("derives hints in the exact C# order with F2 scale and enum-name needs", () => {
    const b = new InMemoryAccessibilityBoard();
    b.setProfile(
      userAccessibilityProfile("u1", [AccessibilityNeed.Visual, AccessibilityNeed.Motor], 1.5, true, true, true),
    );
    assert.equal(b.getProfile("u1")?.textScale, 1.5);
    assert.deepEqual(
      b.hintsFor("u1").map((h) => [h.kind, h.value]),
      [
        ["contrast", "high"],
        ["motion", "reduced"],
        ["aria", "verbose"],
        ["text-scale", "1.50"],
        ["need", "Visual"],
        ["need", "Motor"],
      ],
    );
  });

  it("omits flags that are off and omits text-scale when <= 1", () => {
    const b = new InMemoryAccessibilityBoard();
    b.setProfile(userAccessibilityProfile("u2", [AccessibilityNeed.Hearing], 1.0, false, false, false));
    assert.deepEqual(
      b.hintsFor("u2").map((h) => [h.kind, h.value]),
      [["need", "Hearing"]],
    );
    assert.equal(AccessibilityNeed.Visual, 0);
    assert.equal(AccessibilityNeed.Speech, 4);
  });

  it("domain context exposes prompt + compliance + tools", () => {
    assert.ok(AccessibilityDomainContext.systemPromptSnippet.includes("[DOMAIN: Accessibility]"));
    assert.deepEqual(AccessibilityDomainContext.complianceFlags, ["WCAG_2_2", "UNCRPD", "Equality_Act", "POPIA"]);
    assert.deepEqual(AccessibilityDomainContext.suggestedTools, [
      "screen_reader_test",
      "document_editor",
      "web_audit",
      "analytics",
    ]);
  });
});
